cloud/scope/machinepoolmachine.go

```diff
@@ -318,18 +320,46 @@ func (m *MachinePoolMachineScope) HasLatestModelApplied(_ context.Context, insta
 	// Get the image from the disk URL path to compare with the latest image name
 	diskImage, err := url.Parse(instance.SourceImage)
 	if err != nil {
+		log.Error(err, "Error Parsing the Instance.SourceImage from the disk attached.")
 		return false, err
 	}
 	instanceImage := path.Base(diskImage.Path)
 
-	// Check if the image is the latest
-	if image == instanceImage {
+	// Check if the image is the latest and additionalLabels
+	hasAdditionalLabelsDiff, err := m.hasAdditionalLabelsDiff(ctx, labels)
+	if err != nil {
+		log.Error(err, "Error checking the AdditonalLabels")
+		return false, err
+	}
+	if image == instanceImage && hasAdditionalLabelsDiff {
+		log.Info("Instance Image and AdditionalLabels are same as the ones specified in GCPMachinePool Spec, setting LatestModelApplied to true", "GCPMachinePoolMachine", m.GCPMachinePoolMachine)
 		return true, nil
 	}
 
+	log.Info("One of either Instance Image and AdditionalLabels are not the same as the ones specified in GCPMachinePool Spec, setting LatestModelApplied to false", "GCPMachinePoolMachine", m.GCPMachinePoolMachine)
 	return false, nil
 }
 
+// hasAdditionalLabelsDiff Checks if the Labels applied to the instance are the latest as in the Instance Template.
+// two keys of `capg-role` and `capg-cluster-<CLUSTER-NAME>` as they are added by default by CAPG
+// ref: https://github.com/newrelic-forks/cluster-api-provider-gcp/blob/ef2e7f1e64ebeeb5389c446fe4cf89026fcb8a8a/cloud/services/compute/instances/reconcile_test.go#L244-L24
```

Comment:
Could you please add a one-liner to say we need to ignore the 2 labels?

Reply:
just added a change here https://github.com/newrelic-forks/cluster-api-provider-gcp/pull/25/commits/761d5d79ea1bc944daf93a0a4fbab5af0249208b, please check